magicctapipe/scripts/lst1_magic/semi_automatic_scripts/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,740 @@
+            f"    ls *.gz > {target_dir}/{source_name}/DL1/MC/{particle_type}/$lineB/list_dl0.txt\n",
+            '    string=$lineA"/"\n',
+            f"    export file={target_dir}/{source_name}/DL1/MC/{particle_type}/$lineB/list_dl0.txt\n\n",
+            "    cat $file | while read line; do echo $string${line}"
+            + f" >>{target_dir}/{source_name}/DL1/MC/{particle_type}/$lineB/list_dl0_ok.txt; done\n\n",
+            '    echo "folder $lineB  and node $lineA"\n',
+            f'done 3<"{target_dir}/{source_name}/list_nodes_{particle_type}_complete.txt" 4<"{target_dir}/{source_name}/list_folder_{particle_type}.txt"\n',
+            "",
+        ]
+        f.writelines(lines_of_config_file)
+
+    ################################################################################################################
+    # bash script that applies lst1_magic_mc_dl0_to_dl1.py to all MC data files.
+    ################################################################################################################
+
+    number_of_nodes = glob.glob(f"{MC_path}/node*")
+    number_of_nodes = len(number_of_nodes) - 1
+
+    with open(f"linking_MC_{particle_type}_paths_r.sh", "w") as f:
+        lines_of_config_file = [
+            "#!/bin/sh\n\n",
+            "#SBATCH -p xxl\n",
+            f"#SBATCH -J {process_name}\n",
+            f"#SBATCH --array=0-{number_of_nodes}%50\n",
+            "#SBATCH --mem=10g\n",
+            f"#SBATCH --output={target_dir}/{source_name}/DL1/MC/{particle_type}/slurm-%x.%A_%a.out"
+            f"#SBATCH --error={target_dir}/{source_name}/DL1/MC/{particle_type}/slurm-%x.%A_%a.err"
+            "#SBATCH -n 1\n\n",
+            "ulimit -l unlimited\n",
+            "ulimit -s unlimited\n",
+            "ulimit -a\n",
+            f"cd {target_dir}/{source_name}/DL1/MC/{particle_type}\n\n",
+            f"export INF={target_dir}/{source_name}\n",
+            f"SAMPLE_LIST=($(<$INF/list_folder_{particle_type}.txt))\n",
+            "SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n",
+            "cd $SAMPLE\n\n",
+            f"export LOG={target_dir}/{source_name}/DL1/MC/{particle_type}"
+            + "/simtel_{$SAMPLE}_${SLURM_ARRAY_JOB_ID}_${SLURM_ARRAY_TASK_ID}_all.log\n",
+            "cat list_dl0_ok.txt | while read line\n",
+            "do\n",
+            f"    cd {target_dir}/{source_name}/../\n",
+            f"    conda run -n {env_name} lst1_magic_mc_dl0_to_dl1 --input-file $line --output-dir {target_dir}/{source_name}/DL1/MC/{particle_type}/$SAMPLE --config-file {target_dir}/{source_name}/config_DL0_to_DL1.yaml --focal_length_choice {focal_length}>>$LOG 2>&1\n\n",
+            "done\n",
+            "",
+        ]
+        f.writelines(lines_of_config_file)
+
+
+def lists_and_bash_gen_MAGIC(
+    target_dir, telescope_ids, MAGIC_runs, source, env_name, NSB_match
+):
+
+    """
+    Below we create a bash script that links the the MAGIC data paths to each subdirectory.
```

Comment:
the the ==> the

Reply:
fixed

Reply:
I still see "the the MAGIC data path" in the line above

Reply:
Because I did not push the commit :woman_facepalming: